src/cgw_ucentral_messages_queue_manager.rs

```diff
@@ -0,0 +1,360 @@
+            }
+            if device_msg_queue
+                .get_item(MESSAGE_QUEUE_CONFIGURE_MSG_INDEX)
+                .unwrap()
+                .parsed_message
+                == default_msg
+            {
+                queue_size -= 1;
+            }
+        }
+
+        queue_size
+    }
+
+    // Dequeue messages accoring to priority:
+    // 1. Check if reboot message exist in queue and return it if true
+    // 2. Check if configure message exist in queue and return it if true
+    // 3. Remove other message from qeueu and return it
+    pub async fn dequeue_device_message(
+        &self,
+        device_mac: &String,
+    ) -> Option<CGWUCentralMessagesQueueItem> {
+        let ret_msg: CGWUCentralMessagesQueueItem;
+        let default_msg = CGWUCentralCommand::default();
+        let container_lock = self.queue.read().await;
+
+        if self.get_device_messages_queue_len(device_mac).await == 0 {
+            return None;
+        }
+
+        let mut device_msg_queue = container_lock.get(device_mac).unwrap().write().await;
+        let reboot_msg = device_msg_queue
+            .get_item(MESSAGE_QUEUE_REBOOT_MSG_INDEX)
+            .unwrap()
+            .clone();
+        let configure_msg = device_msg_queue
+            .get_item(MESSAGE_QUEUE_CONFIGURE_MSG_INDEX)
+            .unwrap()
+            .clone();
+
+        if reboot_msg.parsed_message != default_msg {
+            ret_msg = reboot_msg;
+            device_msg_queue.remove_item(MESSAGE_QUEUE_REBOOT_MSG_INDEX);
+            device_msg_queue.insert_item(
+                MESSAGE_QUEUE_REBOOT_MSG_INDEX,
+                CGWUCentralMessagesQueueItem::default(),
+            );
+        } else if configure_msg.parsed_message != default_msg {
+            ret_msg = configure_msg;
+            device_msg_queue.remove_item(MESSAGE_QUEUE_CONFIGURE_MSG_INDEX);
+            device_msg_queue.insert_item(
+                MESSAGE_QUEUE_CONFIGURE_MSG_INDEX,
+                CGWUCentralMessagesQueueItem::default(),
+            );
+        } else {
+            ret_msg = device_msg_queue
+                .remove_item(MESSAGE_QUEUE_OTHER_MSG_INDEX)
+                .unwrap();
+        }
+
+        Some(ret_msg)
+    }
+
+    pub async fn get_device_queue_state(
+        &self,
+        device_mac: &String,
+    ) -> CGWUCentralMessagesQueueState {
+        let container_lock = self.queue.read().await;
+        let device_msg_queue = container_lock.get(device_mac).unwrap().read().await;
+
+        device_msg_queue.get_state()
+    }
+
+    pub async fn set_device_queue_state(
+        &self,
+        device_mac: &String,
+        state: CGWUCentralMessagesQueueState,
+    ) {
+        let container_lock = self.queue.read().await;
+        let mut device_msg_queue = container_lock.get(device_mac).unwrap().write().await;
+
+        device_msg_queue.set_state(state);
+    }
+
+    pub async fn insert_disconnected_device_timeout(&self, device_mac: String, seconds: u64) {
```

Comment:
We can maybe rename this and change the naming scheme whatsoever for all these x_timeout functions: since connection processor will be responsible for calling this function, we can rename this to something simple like:
device_connected
device_disconnected

To notify **queue manager** that this particular device either connected or disconnected - since the **inherent state** has changed for this mac, the **queue manager** is responsible to react accordingly.